src/facade/reply_capture.h

```diff
@@ -47,8 +49,8 @@ class CapturingReplyBuilder : public RedisReplyBuilder {
  private:
   using Error = std::pair<std::string, std::string>;  // SendError (msg, type)
   using Null = std::nullptr_t;                        // SendNull or SendNullArray
-  struct SimpleString : public std::string {};        // SendSimpleString
-  struct BulkString : public std::string {};          // SendBulkString
+  using SimpleString = std::string;                   // SendSimpleString
+  using BulkString = std::string;                     // SendBulkString
```

Comment:
The downside of typedefing is that they are now the same type, and therefore may be mixed / interchanged.
But since we're not supposed to inherit from `std` types, and there's no ABI compatibility constraints, so I guess I'm ok with it :)

Reply:
I was half-wondering what the intention was here. If it was to provide a library-specific type alias (which I assumed) a using is more appropriate. If the intention was to prevent implicit conversions to/from std::string there are more idiomatic ways to do that than inheritance, especially from std classes like you mentioned.

Reply:
It seems that this inheritance was not just a type alias but was also a bit structural.
https://github.com/dragonflydb/dragonfly/actions/runs/5542747532/jobs/10118034255?pr=1538
```
/__w/dragonfly/dragonfly/src/facade/reply_capture.cc:167:8: error: class member cannot be redeclared
  void operator()(const CapturingReplyBuilder::BulkString& bs) {
       ^
/__w/dragonfly/dragonfly/src/facade/reply_capture.cc:163:8: note: previous definition is here
  void operator()(const CapturingReplyBuilder::SimpleString& ss) {
       ^
```